sklearn/neighbors/_base.py

```diff
@@ -385,6 +386,17 @@ def _radius_neighbors_from_graph(graph, radius, return_distance):
 
 class NeighborsBase(MultiOutputMixin, BaseEstimator, metaclass=ABCMeta):
     """Base class for nearest neighbors estimators."""
+    _parameter_constraints = {
+        "n_neighbors": [Interval(Integral, 1, None, closed="left")],
+        "radius": [Interval(Integral, 1, None, closed="left"), None],
+        "weights": [StrOptions({"uniform", "distance"}), callable],
+        "algorithm": [StrOptions({"auto", "ball_tree", "kd_tree", "brute"}), callable],
+        "leaf_size": [Interval(Integral, 1, None, closed="left")],
+        "p": [Interval(Integral, 1, None, closed="left")],
+        "metric": [StrOptions({"minkowski"}), callable],
+        "metric_params": [dict, None],
+        "n_jobs": [Interval(Integral, -1, None, closed="left"), None],
```

Comment:
About this parameter, there is discussion here: https://github.com/scikit-learn/scikit-learn/pull/23491#discussion_r885543391.
I think it can be: 
```suggestion
        "n_jobs": [None, Integral],
```